src/lib/port-finder.ts

```diff
@@ -26,16 +27,28 @@ class PortFinder {
 
 	#isPortFree(): Promise< boolean > {
 		return new Promise( ( resolve ) => {
-			const server = http.createServer();
+			// First try to connect to the port
+			const socket = new net.Socket();
```

Comment:
The previous `isPortFree()` implementation used `http.createServer().listen()` to check port availability. However, this was leading to false positives. The port was showing free when it was actually in use. This might be happening because listen() may not immediately fail if the port is in a TIME_WAIT state or due to OS-level race conditions.

The updated logic first attempts to connect to the port using a socket to improve reliability. A successful connection confirms the port is in use. If the connection fails, we try to bind to the port to to make sure it's truly available.